### DISCUSSION TITLE
Defining templates for NX integrates repos

### DISCUSSION THREAD

**Discussion by eugeniosegala** [nrwl/nx#17559] [General]
Hey,

I'm using an integrated repo containing 5 packages.

Every time I run the usual `nx generate @nx/js:library...` a new package gets created with the usual `project.json` and other files.

I'm in a situation where I'm constalty adding the following object:

```
"version": {
   "executor": "@jscutlery/semver:version",
   "options": {
      "preset": "conventional",
      "tagPrefix": "${projectName}@",
   },
   "configurations": {
      "dry-run": {
         "dryRun": true,
         "verbose": true
      }
   }
},
``` 

To the `project.json` of the newly created package.

I was wondering:

Is there a way to define a pre-set of `targets`, so I don't need to manually add them to every package I create? A sort of template basically.

Alternatively, Is there a way to have all repos relying on common `project.json` so I can delete it from all projects and config everything from a centralised place? This would also help when it comes to update the `targets` globally.

Any ideas?

**Comment by bulldog98** [comment]
Your best bet is to use a [local generator](https://nx.dev/plugins/recipes/local-generators) and reuse the @nx/js:library on and after that add the boilerplate per code.
